app/metrics/numofrequests.go

```diff
@@ -0,0 +1,118 @@
+package metrics
+
+import (
+	"github.com/gorilla/websocket"
+	"github.com/koding/websocketproxy"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/sirupsen/logrus"
+	"github.com/v3io/sidecar-proxy/app/utils"
+	"net/http"
+	"net/http/httputil"
+	"net/url"
+)
+
+var (
+	WebsocketUpgrader = utils.ExtendedWebSocket{
+		WebsocketUpgrader: websocket.Upgrader{
+			ReadBufferSize:  1024,
+			WriteBufferSize: 1024,
+		}}
+)
+
+type NumOfRequestsMetricsHandler struct {
```

Comment:
There's a `MetricHandler` exported factory so:
- Specific metric handlers like `NumOfRequestsMetricsHandler` should not be exported (-> `numOfRequestsMetricsHandler)
- Their factory method (constructor) would return an `MetricHandler` interface and not a pointer to their specific handler struct (i.e. their inner type will be abstracted away from the caller)